### PR TITLE
refactor remix-debug tests

### PR DESCRIPTION
**Pull request by yann300**
to use remix-simulator instead of directly the VM

### CODE REVIEW

File: libs/remix-lib/src/web3Provider/web3VmProvider.ts [hunk 1]
```diff
@@ -107,26 +107,31 @@ export class Web3VmProvider {
   async txWillProcess (data) {
     this.incr++
     this.processingHash = bufferToHex(data.hash())
+
```
Comment (Aniket-Engg) [inline libs/remix-lib/src/web3Provider/web3VmProvider.ts:110]: This file has spaces as changes

File: libs/remix-simulator/src/vm-context.ts [hunk 2]
```diff
@@ -24,31 +24,27 @@ class StateManagerCommonStorageDump extends StateManager {
...
-        const stream = trie.createReadStream()
-        stream.on('data', (val) => {
-          const value = rlp.decode(val.value)
-          storage['0x' + val.key.toString('hex')] = {
-            key: this.keyHashes[val.key.toString('hex')],
-            value: '0x' + value.toString('hex')
-          }
+  async dumpStorage (address): Promise<StorageDump> {
+    return new Promise((resolve, reject) => {
+      this._getStorageTrie(address)
```
Comment (Aniket-Engg) [inline libs/remix-simulator/src/vm-context.ts:29]: Any specific reason to remove `await`

File: libs/remix-simulator/src/vm-context.ts [hunk 3]
```diff
@@ -60,18 +56,18 @@ class StateManagerCommonStorageDump extends StateManager {
...
 
   async setStateRoot (stateRoot) {
-    await this._cache.flush()
-
-    if (stateRoot === this._trie.EMPTY_TRIE_ROOT) {
-      this._trie.root = stateRoot
-      this._cache.clear()
-      this._storageTries = {}
-      return
+    console.log('setstateroot', stateRoot.toString('hex'))
```
Comment (Aniket-Engg) [inline libs/remix-simulator/src/vm-context.ts:59]: extra console

File: libs/remix-simulator/src/vm-context.ts (hunk 2 above)
Comment (yann300) [reply libs/remix-simulator/src/vm-context.ts:29]: I wanted to be as close as possible to the original `dumpStorage` https://github.com/ethereumjs/ethereumjs-monorepo/blob/501be8da11b07d138bb4497f8a18bdf0fa2d11d7/packages/vm/src/state/stateManager.ts#L519

File: libs/remix-debug/test/debugger.ts [hunk 4]
```diff
@@ -259,13 +256,15 @@ function testDebugging (debugManager) {
...
+    const tested = JSON.parse('{"proposalNames":{"value":[{"value":"0x48656C6C6F20576F726C64210000000000000000000000000000000000000000","type":"bytes32"}],"length":"0x1","type":"bytes32[]","cursor":1,"hasNext":false},"p":{"value":"45","type":"uint256"},"addressLocal":{"value":"0x5B38DA6A701C568545DCFCB03FCB875F56BEDDC4","type":"address"},"i":{"value":"2","type":"uint256"},"proposalsLocals":{"value":[{"value":{"name":{"value":"0x48656C6C6F20576F726C64210000000000000000000000000000000000000000","type":"bytes32"},"voteCount":{"value":"0","type":"uint256"}},"type":"struct Ballot.Proposal"}],"length":"0x1","type":"struct Ballot.Proposal[]"}}')
     try {
       const address = debugManager.traceManager.getCurrentCalledAddressAt(327)
       const location = await debugManager.sourceLocationFromVMTraceIndex(address, 327)
       debugManager.decodeLocalsAt(327, location, (error, decodedlocals) => {
         if (error) return t.end(error)
-        t.ok(deepequal(decodedlocals, tested), `locals does not match. expected: ${JSON.stringify(tested)} - current: ${decodedlocals}`)
+        const res = deepequal(decodedlocals, tested)
+        console.log(res)
```
Comment (Aniket-Engg) [inline libs/remix-debug/test/debugger.ts:266]: Is this console needed?

File: libs/remix-debug/test/decoder/stateTests/mapping.ts [hunk 5]
```diff
@@ -2,57 +2,62 @@ import { compilerInput } from '../../helpers/compilerHelper'
...
 import { compile } from 'solc'
 import * as stateDecoder from '../../../src/solidity-decoder/stateDecoder'
-import { sendTx, initVM } from '../vmCall'
+import * as vmCall from '../../vmCall'
 import { StorageResolver } from '../../../src/storage/storageResolver'
 import { StorageViewer } from '../../../src/storage/storageViewer'
+import {  Address, bufferToHex } from 'ethereumjs-util'
 
 module.exports = async function testMappingStorage (st, cb) {
+  console.log('start testMappingStorage')
```
Comment (Aniket-Engg) [inline libs/remix-debug/test/decoder/stateTests/mapping.ts:11]: and this one

File: libs/remix-debug/test/decoder/stateTests/mapping.ts [hunk 6]
```diff
@@ -2,57 +2,62 @@ import { compilerInput } from '../../helpers/compilerHelper'
...
   var mappingStorage = require('../contracts/mappingStorage')
-  var privateKey = Buffer.from('dae9801649ba2d95a21e688b56f77905e5667c44ce868ec83f82e838712a2c7a', 'hex')
-  var vm = await initVM(st, privateKey)
+  var privateKey = Buffer.from('503f38a9c967ed597e47fe25643985f032b072db8075426a92110f82df48dfcb', 'hex')
   var output = compile(compilerInput(mappingStorage.contract))
-  output = JSON.parse(output)
-  sendTx(vm, {nonce: 0, privateKey: privateKey}, null, 0, output.contracts['test.sol']['SimpleMappingState'].evm.bytecode.object, function (error, data) {
+  output = JSON.parse(output);
+  console.log('BYTECODE', output.contracts['test.sol']['SimpleMappingState'].evm.bytecode.object);
```
Comment (Aniket-Engg) [inline libs/remix-debug/test/decoder/stateTests/mapping.ts:16]: This one too

File: libs/remix-debug/test/decoder/stateTests/mapping.ts [hunk 7]
```diff
@@ -2,57 +2,62 @@ import { compilerInput } from '../../helpers/compilerHelper'
...
-      const txHash = data.hash
-      vm.web3.eth.getTransaction(txHash, (error, tx) => {
+      web3.eth.getTransactionReceipt(hash, (error, tx) => {
         if (error) {
           console.log(error)
           st.end(error)
         } else {
-          testMapping(st, vm, privateKey, tx.contractAddress, output, cb)
+          // const storage = await this.vm.stateManager.dumpStorage(data.to)
+          console.log(tx);
```
Comment (Aniket-Engg) [inline libs/remix-debug/test/decoder/stateTests/mapping.ts:29]: This one

File: libs/remix-debug/test/vmCall.ts [hunk 8]
```diff
@@ -1,63 +1,52 @@
...
-  } catch (error) {
-    console.log(error)
-  } 
-
-  var web3Provider = new remixLib.vm.Web3VMProvider()
-  web3Provider.setVM(vm)
-  vm.web3 = web3Provider
-  return vm
+    cb = cb || (() => {})
+    console.log('TX', {
```
Comment (Aniket-Engg) [inline libs/remix-debug/test/vmCall.ts:27]: This one

File: libs/remix-debug/test/vmCall.ts [hunk 9]
```diff
@@ -1,63 +1,52 @@
...
+      gas: 7000000
+    })
+    const receipt = await web3.eth.sendTransaction({
+      from: utileth.Address.fromPrivateKey(from.privateKey).toString('hex'),
+      to,
+      value,
+      data,
+      gas: 7000000
+    })
+    console.log('end tx')
```
Comment (Aniket-Engg) [inline libs/remix-debug/test/vmCall.ts:41]: this one

File: libs/remix-simulator/src/vm-context.ts (hunk 2 above)
Comment (Aniket-Engg) [reply libs/remix-simulator/src/vm-context.ts:29]: ok

**Review by Aniket-Engg — APPROVED** [review]

**Pull request merged**